src/App.jsx

```diff
@@ -14,45 +17,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const buttons = [
+  {
+    name: 'Sort alphabetically',
+    class: 'is-info',
+  },
+  {
+    name: 'Sort by length',
+    class: 'is-success',
+  },
+];
+
+const sortList = (goods, buttonClass, isReversed) => {
+  const list = [...goods];
+  const sortLength = (good1, good2) => good1.length - good2.length;
+
+  if (buttonClass === 'is-info') {
+    return isReversed ? list.sort().reverse() : list.sort();
+  }
+
+  if (buttonClass === 'is-success') {
+    return isReversed
+      ? list.sort(sortLength).reverse()
+      : list.sort(sortLength);
+  }
+
+  if (isReversed) {
+    list.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return list;
+};
+
+export const App = () => {
+  const [status, setStatus] = useState('');
+  const [direction, setDirection] = useState(false);
+
+  const preparedGoods = sortList(goodsFromServer, status, direction);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {buttons.map(button => (
+          <button
+            type="button"
+            key={button.class}
+            className={cn(
+              'button',
+              `${button.class}`, {
+                'is-light': status !== button.class,
+              },
+            )}
+            onClick={() => {
```

Comment:
Move the onClick logic to a separate function. It follows the best practice of separating concerns and keeping your JSX code clean and focused on rendering components rather than defining complex logic directly within it.

Reply:
I am rendering 2 sort buttons from 1 tag (line 69-85) so for now i don`t know how to make it work if i move logic in separate function. I can rewrite lines 69-85 and use two <button> tags instead but then i`ll double my code.

Reply:
just create function outside of JSX and use its reference here. It won't broke your code